terraso_backend/apps/auth/views.py

```diff
@@ -243,3 +250,68 @@ def terraso_login(request, user):
     dj_login(request, user, backend="django.contrib.auth.backends.ModelBackend")
 
     return access_token, refresh_token
+
+
+class TokenExchangeView(View):
+    @staticmethod
+    def _create_or_fetch_user(
+        email: str = "",
+        given_name: str = "",
+        family_name: str = "",
+        picture: Optional[str] = None,
+        **kwargs,
+    ):
+        account_service = AccountService()
+        additional_kwargs = {}
+        if picture:
+            additional_kwargs["profile_image_url"] = picture
+        # TODO: using a private method of AccountService is weird, should be refactored
+        # Should be a public method, and arguably static
+        user, created = account_service._persist_user(
+            email, first_name=given_name, last_name=family_name, **additional_kwargs
+        )
+        return user, created
+
+    @staticmethod
+    def _check_request(contents):
+        missing_keys = []
+        for key in ("jwt", "provider"):
+            if key not in contents:
+                missing_keys.append(key)
+        if missing_keys:
+            return JsonResponse({"missing_keys": missing_keys}, status=400)
+        provider = contents["provider"]
+        if provider not in settings.JWT_EXCHANGE_PROVIDERS:
+            return JsonResponse({"bad_provider": provider}, status=400)
+
+    @staticmethod
+    def _token_error(e):
+        match e.error_type:
+            case "jwks_error" | "bad_config":
```

Comment:
issue: this looks like a bitwise or, which is probably not what you want for strings.

Reply:
This is actually part of [Python pattern matching](https://peps.python.org/pep-0636/#or-patterns)